File: AdminArea/AdminArea.Web/Modules/onlinestore/diagram/diagramForm.cs
using Serenity.ComponentModel;

namespace AdminArea.onlinestore.Forms;

[FormScript("onlinestore.diagram")]
[BasedOnRow(typeof(diagramRow), CheckNames = true)]
public class diagramForm
{
    public string Name { get; set; }
    public int PrincipalId { get; set; }
    public int Version { get; set; }
    public byte[] Definition { get; set; }
}

File: AdminArea/AdminArea.Web/Modules/ServerTypes/onlinestore/diagramForm.ts
import { StringEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface diagramForm {
    Name: StringEditor;
    PrincipalId: IntegerEditor;
    Version: IntegerEditor;
    Definition: StringEditor;
}

export class diagramForm extends PrefixedContext {
    static formKey = 'onlinestore.diagram';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!diagramForm.init)  {
            diagramForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(diagramForm, [
                'Name', w0,
                'PrincipalId', w1,
                'Version', w1,
                'Definition', w0
            ]);
        }
    }
}